### PR TITLE
add sleep parameter to batchman

### PR DESCRIPTION
**Pull request by SimplyAhmazing**
This PR allows the `batch` function to take a sleep argument. This argument throttles the fetching of batches by sleeping `sleep` seconds when the `batch` method is called.

### CODE REVIEW

File: R/batch.R [hunk 1]
```diff
@@ -13,10 +13,20 @@
...
+    batch_fn,
+    keys,
+    splitting_strategy = NULL,
+    combination_strategy = batchman::combine,
+    size = 50,
+    trycatch = FALSE,
+    batchman.verbose = isTRUE(interactive()),
+    stop = FALSE,
+    retry = 0,
+    sleep=0
```
Comment (peterhurford) [inline R/batch.R]: Could you put space around the ` = `?

File: R/batch.R [hunk 2]
```diff
@@ -46,6 +56,8 @@ batch <- function(batch_fn, keys, splitting_strategy = NULL,
           )
         } else { eval(new_call, envir = run_env) }
 
+        if (sleep > 0) { Sys.sleep(sleep) }
```
Comment (peterhurford) [inline R/batch.R]: :+1:

File: tests/testthat/test-batch.R [hunk 3]
```diff
@@ -340,3 +341,52 @@ lapply(
...
+  {
+    env <- list2env(list(called=FALSE))
+    package_stub(
+      "base",
+      "Sys.sleep",
+      function(...) {
+        env$called <- TRUE
+      },
+      {
+        batch_callable <- batch(
```
Comment (peterhurford) [inline tests/testthat/test-batch.R]: I've been calling these `batch_fn` but this is fine.

**Comment by peterhurford** [comment]
@SimplyAhmazing Also, Travis is failing.  This is because the new dependency, testthatsomemore, needs to be added to [the travis.yml file](https://github.com/peterhurford/batchman/blob/master/.travis.yml).

File: R/batch.R (hunk 1 above)
Comment (SimplyAhmazing) [inline R/batch.R]: Done

**Comment by SimplyAhmazing** [comment]
.travis.yml contains this line,

`- "./travis-tool.sh install_r R6 testthatsomemore"` -- does this not install testthatsomemore

**Comment by peterhurford** [comment]
@SimplyAhmazing `package_stub` isn't in the CRAN version of `testthatsomemore`, so we need to install it from GitHub -- `robertzk/testthatsomemore`.

**Comment by SimplyAhmazing** [comment]
Sweet!!

**Pull request merged**